contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2566,6 +2615,52 @@ sp_dropserver_internal(PG_FUNCTION_ARGS)
 	return (Datum) 0;
 }
 
+Datum
+sp_serveroption_internal(PG_FUNCTION_ARGS)
+{
+	char *servername = PG_ARGISNULL(0) ? NULL : lowerstr(text_to_cstring(PG_GETARG_VARCHAR_PP(0)));
+	char *optionname = PG_ARGISNULL(1) ? NULL : lowerstr(text_to_cstring(PG_GETARG_VARCHAR_PP(1)));
+	char *optionvalue = PG_ARGISNULL(2) ? NULL : lowerstr(text_to_cstring(PG_GETARG_VARCHAR_PP(2)));
+
+	if(!pltsql_enable_linked_servers)
+		ereport(ERROR,
+			(errcode(ERRCODE_FDW_ERROR),
+				errmsg("'sp_serveroption' is not currently supported in Babelfish")));
+
+	if (servername == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_FDW_ERROR),
+				 errmsg("@server parameter cannot be NULL")));
+
+	if (optionname == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_FDW_ERROR),
+				 errmsg("@optname parameter cannot be NULL")));
+
+	if (optionvalue == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_FDW_ERROR),
+				 errmsg("@optvalue parameter cannot be NULL")));
+
+	if (optionname && strlen(optionname) == 13 && strncmp(optionname, "query timeout", 13) == 0)
+		update_bbf_server_options(servername, atoi(optionvalue), false);
```

Comment:
Should we error out if atoi(optionvalue) is negative? We should see what values are acceptable in T-SQL

Reply:
in TSQL, we get an error when the option value is negative. I have added the support for the same.